src/plugins/Modal.js

```diff
@@ -12,160 +12,104 @@ export default class Modal extends Plugin {
+    this.core.setState({modal: modal})
   }
 
-  render () {
-    // http://dev.edenspiekermann.com/2016/02/11/introducing-accessible-modal-dialog
-
-    return `
-      <div class="UppyModal"
-           aria-hidden="true"
-           aria-labelledby="modalTitle"
-           aria-describedby="modalDescription"
-           role="dialog">
-        <div class="UppyModal-overlay js-UppyModal-close" tabindex="-1"></div>
-        <div class="UppyModal-inner">
-          <button class="UppyModal-close js-UppyModal-close" title="Close Uppy modal">×</button>
-
-          <ul class="UppyModalTabs" role="tablist"></ul>
-
-          <div class="UppyModalContent">
-            <div class="UppyModal-presenter"></div>
-            <div class="UppyModal-progress">
-              <button class="UppyModal-next js-UppyModal-next">${this.core.i18n('upload')}</button>
-              <div class="UppyModal-progressBarContainer"></div>
-            </div>
-          </div>
+  showTabPanel (id) {
+    const modal = this.core.getState().modal
+
+    // hide all panels, except the one that matches current id
+    modal.targets.forEach((target) => {
+      if (target.type === 'acquirer') {
+        if (target.id === id) {
+          target.isHidden = false
+          return
+        }
+        target.isHidden = true
+      }
+    })
 
-        </div>
-      </div>
-    `
+    this.core.setState({modal: modal})
   }
 
   hideModal () {
-    this.isModalVisible = false
-    this.modalEl.setAttribute('aria-hidden', 'true')
-    this.core.emitter.emit('reset')
-  }
-
-  showModal () {
-    this.isModalVisible = true
-    this.modalEl.removeAttribute('aria-hidden')
-    this.initTabs()
-  }
+    const modal = this.core.getState().modal
+    modal.isHidden = true
+    this.core.setState({modal: modal})
 
-  hideAllTabPanels () {
-    this.tabPanels.forEach((tabPanel) => tabPanel.setAttribute('aria-hidden', true))
-    this.tabs.forEach((tab) => tab.removeAttribute('aria-selected'))
+    document.body.style.overflow = 'visible'
   }
 
-  showTabPanel (pluginSelector, pluginName) {
-    this.hideAllTabPanels()
+  showModal () {
+    // this.core.emitter.emit('modal-open')
 
-    const tab = document.querySelector(`[aria-controls="${pluginName}"]`)
-    const tabPanel = document.querySelector(`.${pluginSelector}`)
+    const modal = this.core.getState().modal
+    modal.isHidden = false
 
-    tabPanel.removeAttribute('aria-hidden')
-    tab.setAttribute('aria-selected', 'true')
+    // Show first acquirer plugin when modal is open
+    modal.targets.some((target) => {
+      if (target.type === 'acquirer') {
+        target.isHidden = false
+        return true
+      }
+    })
 
-    this.core.log(pluginName)
-    this.core.getPlugin(pluginName).focus()
+    this.core.setState({modal: modal})
 
-    // Remove `next` buttons provided by plugins, since Modal has its own
-    // document.querySelector('.UppyNextBtn').remove()
+    document.body.style.overflow = 'hidden'
```

Comment:
I'd prefer if we could toggle a class on the body to control this, rather than adding an inline style, but it's not a big deal.